app/workers/unpublish_taxon_worker.rb

```diff
@@ -0,0 +1,24 @@
+class UnpublishTaxonWorker
+  include Sidekiq::Worker
+
+  def perform(taxon_content_id, redirect_to_content_id)
```

Comment:
There are two things happening here. Redirecting the taxon (to the parent taxon?) and re-tagging content to the parent taxon. These two operations seem like they are independent of each other.

To make the code easier to read, I would move the code into two separate functions that are called by `perform`.
Something like `retag_content_to_parent_taxon(taxon_content_id)` and `redirect_taxon(redirect_to_content_id)`.

Reply:
fixed